### PR TITLE
Initial Win support

### PR DESCRIPTION
**Pull request by masahi**
* fixed path handling
* basic Def -> Ref mapping works (enough for pytorch & mxnet)
* there seems to be an issue in import resolver, so Workspace.find_defs doesn't work on Win atm. test_import_resolver.py is disabled on Win for now, some of the tests in test_workspace.py return empty results. Haven't looked into what's going on in PyImportResolver (looks complicated), any thoughts? @tqchen

### CODE REVIEW

File: python/ffi_navigator/langserver.py [hunk 1]
```diff
@@ -10,8 +10,14 @@
 from pyls_jsonrpc import dispatchers, endpoint, streams
 
 
+def is_win():
+    return sys.platform == "win32"
+
```
Comment (comaniac) [inline python/ffi_navigator/langserver.py]: Is there any reason for this function be be implemented twice (also in .util)?

File: python/ffi_navigator/util.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+def is_win():
+    return sys.platform == "win32"
+
+
+def join_path(root_path, relative_path):
+    return str(root_path / Path(relative_path))
+
+
+def normalize_path(raw_path):
```
Comment (comaniac) [inline python/ffi_navigator/util.py]: `normalize_path` looks a bit weird. Something like `unify_path` might be better?

File: python/ffi_navigator/util.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import sys
+from pathlib import Path
+
+
+def is_win():
+    return sys.platform == "win32"
+
+
+def join_path(root_path, relative_path):
```
Comment (comaniac) [inline python/ffi_navigator/util.py]: docstring

**Comment by tqchen** [comment]
Would be great if we can figure out what is going on in the import resolver as well. Likely all have things to do with windows path vs the linux one

**Comment by masahi** [comment]
Does import resolver need to care about the platform difference? It uses os.path.abspath inside, and it prepends windows specific prefix like D:\tvm. Maybe it assumes that the path starts with "\"?

File: python/ffi_navigator/util.py (hunk 2 above)
Comment (masahi) [reply python/ffi_navigator/util.py]: https://docs.python.org/3/library/os.path.html#os.path.normpath

Comment (comaniac) [reply python/ffi_navigator/util.py]: Gotcha. Then I'm fine with both `normalize_path`.

**Comment by tqchen** [comment]
the D:\ prefix should be fine, but perhaps there might be issues with "/" vs "\" somewhere, as the code uses "/" for most of the separators.

**Comment by masahi** [comment]
@tqchen @comaniac it's fixed now. All tests work on Win. The problem was os.path.abspath adds D:\\ prefix, but os.path.join doesn't, so this inconsistency was causing some dict lookup to fail.

ready for merge.

**Comment by masahi** [comment]
that tag is from tvm repo, it's not too old so no need to update frequently.

**Comment by masahi** [comment]
better yet, we should have a dummy repo for tvm too, to actually test the output. Compared to mxnet and pytorch, tvm dialect is complicated, so it is not straightforward to setup the repo.

**Comment by comaniac** [comment]
I am actually encoutering the same issue in my project which depends on a specific TVM commit. I modified the TVM docker install script to create a docker image to clone TVM, checkout a specific commit, and build for testing. In general, we should improve the TVM building experience, but this is out of scope of this discussion.

**Comment by tqchen** [comment]
Let us just create a tvm dummy repo as well to run the unittestcases

**Pull request merged**